docs/user/monitors.rst

```diff
@@ -0,0 +1,60 @@
+.. _monitors:
+
+========
+Monitors
+========
+
+Monitors are components that monitor the target for specific behaviour. A
+monitor can be passive and just observe and provide data or behave more active
+interacting directly with the target. Some monitors also have the capability to
+start, stop and restart targets.
+
+Detecting a crash or misbehaviour of your target can be a complex, non-straight
+forward process depending on the tools you have available on your targets host;
+this holds true especially for embedded devices. Boofuzz provides three main
+monitor implementations:
+
+- *ProcessMonitor*, a Monitor that collects debug info from process on Windows
```

Comment:
Added. I'm still not quite happy with the overall structure of the docs, because the docstrings propagate through the inheritance. Most of them make only sense on the interface level as they specify the expected behavior of the functions.

I thought about adding empty docstrings on functions where I could not think of something to document (because, well, it's only the implementation of the interface, the high-level description is there) but I'm not quite convinced yet.

Reply:
Empty docstrings seem to be the only way to disable the inheritance.
I'd be fine with that if the docstring from the base class doesn't make sense in some places.

Reply:
Quick comment: This is another place where our use of interfaces breaks down -- interfaces are designed to intrinsically define behavior the object does, not a set of behaviors from which an object may implement some but not others. (again, I don't know of something significantly better in this situation)

Reply:
Now with a little distance I think it is more important to only implement what is needed. Therefore, I removed show-inheritance from the doc directives on the monitor page. Still, this doesn't solve the docstring problem, but it reduces it to the methods that actually are implemented.

I'll give them a quick doc and then it should be good.